From 12ce243471d8bf7f2b6a738ee2c7b0fe16266cc2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20=C4=8Cerm=C3=A1k?= <sairon@sairon.cz>
Date: Thu, 16 Nov 2023 12:46:48 +0100
Subject: [PATCH] ARM: dts: bcm2711: yellow: Use mini-UART for Bluetooth by
 default
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Apply changes from miniuart-bt-overlay.dts to Yellow DTS to use
mini-UART for Bluetooth by default and initialize hci device in the
kernel driver (can be overriden by setting dtparam=krnbt=off).
Full UART should be available on J11.

Signed-off-by: Jan Čermák <sairon@sairon.cz>
---
 .../broadcom/bcm2711-rpi-cm4-ha-yellow.dts    | 36 ++++++++++++++-----
 1 file changed, 28 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/broadcom/bcm2711-rpi-cm4-ha-yellow.dts b/arch/arm/boot/dts/broadcom/bcm2711-rpi-cm4-ha-yellow.dts
index b8f0bc2df062d..26033f2d30d0c 100644
--- a/arch/arm/boot/dts/broadcom/bcm2711-rpi-cm4-ha-yellow.dts
+++ b/arch/arm/boot/dts/broadcom/bcm2711-rpi-cm4-ha-yellow.dts
@@ -318,7 +318,7 @@ chosen {
 	};
 
 	aliases {
-		serial0 = &uart1;
+		serial0 = &uart0;
 		serial1 = &uart4;
 		serial2 = &uart5;
 		mmc0 = &emmc2;
@@ -334,6 +334,8 @@ aliases {
 		spi4 = &spi4;
 		spi5 = &spi5;
 		spi6 = &spi6;
+		bluetooth = &uart1;
+
 		/delete-property/ intc;
 	};
 
@@ -348,12 +350,22 @@ &mmcnr {
 };
 
 &uart0 {
-	pinctrl-0 = <&uart0_pins &bt_pins>;
+	pinctrl-0 = <&uart0_pins>;
 	status = "okay";
 };
 
 &uart1 {
-	pinctrl-0 = <&uart1_pins>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart1_bt_pins>;
+	status = "okay";
+};
+
+&bt {
+	status = "disabled";
+};
+
+&minibt {
+	status = "okay";
 };
 
 &spi0 {
@@ -490,15 +502,21 @@ bt_pins: bt_pins {
 	};
 
 	uart0_pins: uart0_pins {
+		brcm,pins;
+		brcm,function;
+		brcm,pull;
+	};
+
+	uart1_pins: uart1_pins {
 		brcm,pins = <32 33>;
-		brcm,function = <BCM2835_FSEL_ALT3>;
+		brcm,function = <BCM2835_FSEL_ALT5>; /* alt5=UART1 */
 		brcm,pull = <0 2>;
 	};
 
-	uart1_pins: uart1_pins {
-		brcm,pins;
-		brcm,function;
-		brcm,pull;
+	uart1_bt_pins: uart1_bt_pins {
+		brcm,pins = <32 33 30 31>;
+		brcm,function = <BCM2835_FSEL_ALT5>; /* alt5=UART1 */
+		brcm,pull = <0 2 2 0>;
 	};
 
 	uart2_pins: uart2_pins {
@@ -683,5 +701,7 @@ __overrides__ {
 		cam1_reg = <&cam1_reg>,"status";
 		cam1_reg_gpio = <&cam1_reg>,"gpio:4",
 				  <&cam1_reg>,"gpio:0=", <&gpio>;
+
+		krnbt = <&minibt>,"status";
 	};
 };
